crates/common/src/raindex_client/orderbook_yaml.rs

```diff
@@ -0,0 +1,339 @@
+    ///
+    /// Extracts and returns all unique blockchain network chain IDs that are available
+    /// in the current orderbook configuration.
+    ///
+    /// ## Examples
+    ///
+    /// ```javascript
+    /// const result = client.getUniqueChainIds();
+    /// if (result.error) {
+    ///   console.error("Error getting chain IDs:", result.error.readableMsg);
+    ///   return;
+    /// }
+    /// const chainIds = result.value;
+    /// console.log("Available chains:", chainIds);
+    /// ```
+    #[wasm_export(
+        js_name = "getUniqueChainIds",
+        return_description = "Returns a list of unique chain IDs from all available networks.",
+        unchecked_return_type = "number[]"
+    )]
+    pub fn get_unique_chain_ids(&self) -> Result<Vec<u32>, RaindexError> {
+        let networks = self.get_all_networks()?;
+        Ok(networks.values().map(|cfg| cfg.chain_id).collect())
+    }
+
+    /// Retrieves all available networks with their configurations
+    ///
+    /// Returns a comprehensive map of all network configurations available in the orderbook YAML.
+    /// Each entry contains detailed network information including RPC endpoints and chain-specific settings.
+    ///
+    /// ## Examples
+    ///
+    /// ```javascript
+    /// const result = client.getAllNetworks();
+    /// if (result.error) {
+    ///   console.error("Error getting networks:", result.error.readableMsg);
+    ///   return;
+    /// }
+    /// const networks = result.value;
+    /// for (const [key, config] of networks) {
+    ///   console.log(`Network key: ${key}, Chain ID: ${config.chainId}`);
+    /// }
+    /// ```
+    #[wasm_export(
+        js_name = "getAllNetworks",
+        return_description = "Returns a map of all available networks with their configurations. Keys are network names, values are NetworkCfg objects.",
+        unchecked_return_type = "Map<string, NetworkCfg>"
+    )]
+    pub fn get_all_networks(&self) -> Result<HashMap<String, NetworkCfg>, RaindexError> {
+        Ok(self.orderbook_yaml.get_networks()?)
+    }
+
+    /// Retrieves network configuration for a specific chain ID
+    ///
+    /// Finds and returns the network configuration that matches the provided chain ID.
+    /// This is useful when you need to access network-specific settings like RPC URLs.
+    ///
+    /// ## Examples
+    ///
+    /// ```javascript
+    /// const result = client.getNetworkByChainId(1); // Ethereum mainnet
+    /// if (result.error) {
+    ///   console.error("Network not found:", result.error.readableMsg);
+    ///   return;
+    /// }
+    /// const networkConfig = result.value;
+    /// console.log(`Found network: ${networkConfig}`);
+    /// ```
+    #[wasm_export(
+        js_name = "getNetworkByChainId",
+        return_description = "Returns the configuration for a specific network identified by its chain ID",
+        unchecked_return_type = "NetworkCfg"
+    )]
+    pub fn get_network_by_chain_id(
```

Comment:
hm there's actually nothing in the spec that says you can't have two networks in the networks map with the same chain id